cleanhttp.go

```diff
@@ -26,3 +29,11 @@ func DefaultClient() *http.Client {
 		Transport: DefaultTransport(),
 	}
 }
+
+func SetFinalizer(transport *http.Transport) {
+	runtime.SetFinalizer(&transport, FinalizeTransport)
+}
+
+func FinalizeTransport(t **http.Transport) {
+	(*t).CloseIdleConnections()
```

Comment:
Is there any reason to expose this function? why don't we just inline it above into SetFinalizer. I don't see a reason (esp with the weird function signature) for this to exist on its own as a public API.